framework/src/service/LockSet.cpp

```diff
@@ -0,0 +1,49 @@
+/*=============================================================================
+  Library: CppMicroServices
+  Copyright (c) The CppMicroServices developers. See the COPYRIGHT
+  file at the top-level directory of this distribution and at
+  https://github.com/CppMicroServices/CppMicroServices/COPYRIGHT .
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+=============================================================================*/
+
+#include "LockSet.h"
+
+#ifdef _MSC_VER
+#    pragma warning(push)
+#    pragma warning(disable : 4355)
+#endif
+
+namespace cppmicroservices
+{
+
+#ifdef US_ENABLE_THREADING_SUPPORT
+    LockSet::LockSet(std::shared_ptr<ServiceRegistrationBasePrivate> reg,
+                     std::shared_ptr<ServiceRegistrationCoreInfo> coreInfo)
+    {
+        if (reg != nullptr)
+        {
+            regL = reg->Lock();
+        }
+        coreInfoL = coreInfo->Lock();
+    }
+#else
+    LockSet::LockSet(std::shared_ptr<ServiceRegistrationBasePrivate>,
+                     std::shared_ptr<ServiceRegistrationCoreInfo>)
+    {
+    }
+#endif
```

Comment:
I don't understand the purpose of the `LockSet` class. the `Lock()` function already uses mutexes if US_ENABLE_THREADING_SUPPORT is set and doesn't if the macro is unset. This class seems redundant.

Reply:
This is essentially a way of linking the locking of registrationBasePrivate and coreInfo. In the current implementation, the registrationBasePrivate is Locked whenever any data in it is accessed (namely properties, service, etc). That is no longer possible as that data, that originally was in the registrationBasePrivate, is now in CoreInfo. And, as the ServiceRegistrationBasePrivate dies independently of the coreInfo now, we cannot only rely on the locking of the registration and so we must also lock CoreInfo. This was written so that those two can be coupled without continually having to replicate this code all across the code base whenever we need to lock both. 

I am not sure this answered your question, but if it didn't let me know.